Dockerfile

```diff
@@ -118,7 +118,9 @@ ARG RUNTIME_PACKAGES="\
   zlib1g \
   # Barcode splitter
   libzbar0 \
-  poppler-utils"
+  poppler-utils \
+  # XRechnung
+  default-jre"
```

Comment:
I'm not super interested in including a whole Java runtime for something which seems very niche.  This is going to add significant size to the image.

Reply:
I see your concern. I'm not too happy with using the java mustang-cli either. However I decided to use it as it appears to be the easiest and most reliable way of handling those files, as this is a well matured and maintained software.
Maybe we could put the installation in Docker behind a feature-flag aswell?

Otherwise I also could give this library a try: https://github.com/akretion/factur-x to see if it has smaller dependencies.

But are you in general fine with an xml handler which tries to understand the schema and handles it accordingly? I dont know about other countries laws / standards but we could design this parser so, that it could support them aswell in the future.

As the new digital invoice law will be in place beginning of next year in Germany and from 2026 on in France + probably sooner or late the entire EU (its based on an eu-standard: https://ec.europa.eu/digital-building-blocks/sites/display/DIGITAL/eInvoicing) this kind of feature might become more important or interesting for business use-cases in the future.

Reply:
Yes, a separate container like Tika/Gotenberg would be fine. Just defining a basic apo for getting content, dates and rendering

Reply:
@Marcel2508 have you seen the following repositories?  
https://github.com/marrelUnderscore/RechnunglessParser  
https://github.com/marrelUnderscore/RechnunglessConverter  

What @marrelUnderscore did looks like a good approach to me. It uses a separate container to generate the PDF and thumbnail. May we can "finalize" his project and get the parser part merged into paperless.

Reply:
@stumpylog I've refactored the parser to use an external "rechnungless" service by @marrelUnderscore . It can be conditionally enabled and I hope to have added it everywhere necessary (docs, installer, ...)

Reply:
I'll also note an alternative approach that wouldn't require relying on an external tool: https://github.com/paperless-ngx/paperless-ngx/discussions/6823#discussioncomment-11518801

Im not sure which approach is best Im just a little wary of pngx becoming a Frankenstein project that requires many other services / containers.

Also, why are you including plaintext (.txt) in the new changes?

Reply:
Tika undoubtedly already handles xml...

Reply:
Hey, I've had a look into https://github.com/paperless-ngx/paperless-ngx/discussions/6823#discussioncomment-11518801 in the beginning, but I wasn't able to get it working and while researching I found comments about the python xml parse beeing quite slow (dont know if it still applies today). But I think having an external service has the added benefit that we can update it independently of paperless. I'd think the xml schema could change regulary in the future / new schema versions might be added.


I included plaintext .txt because in my observations firefox sends content type text/plain sometimes (?)

Reply:
@shamoon I dont think tikas xml parsing will help us here because it does not know about the "e-invoice" formats and I dont think tika should act on the content of the document but just parse it. 

We could make the rechnungless server more generic - so that in the future it could be used for all kinds of documents / textfiles with meaning / a schema in itself.

Reply:
I’m not convinced that this can’t be done with tika/gotenberg, we’re basically just talking about rendering the xml nicely? I haven’t looked at it, and not saying I am making a demand but if possible it might be worth doing, or at least exploring. Maybe I’ll look into it, I think stumpylog is pretty busy anyway at the moment